pkg/server/server.go

```diff
@@ -424,3 +428,46 @@ func setMasterRoleLabel(ctx context.Context, nodes v1.NodeClient) error {
 	}
 	return nil
 }
+
+
+func setClusterDNSConfig(ctx context.Context, controlConfig *Config, configMap v1.ConfigMapClient) error{
+	for {
+		nodeName := os.Getenv("NODE_NAME")
+		// check if configmap already exists
+		_, err := configMap.Get("kube-system", "cluster-dns", metav1.GetOptions{})
+		if err == nil {
+			logrus.Infof("cluster dns configmap already exists")
+			break
+		}
+		clusterDNS := controlConfig.ControlConfig.ClusterDNS
+		clusterDomain := controlConfig.ControlConfig.ClusterDomain
+		c := &corev1.ConfigMap{
+			TypeMeta:   metav1.TypeMeta{
+				Kind:       "ConfigMap",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:                       "cluster-dns",
+				Namespace:                  "kube-system",
+			},
+			Data:  map[string]string{
+				"clusterDNS": clusterDNS.String(),
+				"clusterDomain": clusterDomain,
+			},
+		}
+		_, err = configMap.Create(c)
+		if err == nil {
+			logrus.Infof("cluster dns configmap has been set successfully")
+			break
+		}
+		logrus.Infof("Waiting for master node %s startup: %v", nodeName, err)
+		time.Sleep(1 * time.Second)
```

Comment:
Looks like this could be removed in favor of the 1 second wait in the select block.